pkg/controller/computenodeopenstack/computenodeopenstack_controller.go

```diff
@@ -395,45 +483,114 @@ func updateNodesStatus(c client.Client, instance *computenodev1alpha1.ComputeNod
+
+						err = addToBeRemovedTaint(kclient, updateNode)
+						if err != nil {
+							return err
+						}
+					}
+				}
+
+				// wait until all pods in openstack namespace got shutdown
+				podsOnNode, err := getRunningReadyPodsOnNode(kclient, node.Name)
 				if err != nil {
 					return err
 				}
+				if podsOnNode > 0 {
+					log.Info(fmt.Sprintf("Still %s openstack pods in Running state: %s", podsOnNode, node.Name))
+					ResyncPeriod = 30 * time.Second
+					return nil
+				}
 
-				for i, nodeToDelete := range instance.Spec.NodesToDelete {
-					if nodeToDelete.Name == node.Name {
-						if len(instance.Spec.NodesToDelete) == 1 {
-							instance.Spec.NodesToDelete = []computenodev1alpha1.NodeToDelete{}
-						} else {
-							instance.Spec.NodesToDelete = removeNode(instance.Spec.NodesToDelete, i)
-						}
-						err = c.Update(context.TODO(), instance)
-						if err != nil {
-							return err
+				// trigger node drain post tasks (new pod)
+				err = triggerNodeDrain(c, node.Name, instance, false, openstackClientAdmin, openstackClient)
+				if err != nil {
+					return err
+				}
+				log.Info("Node drain POST job created!")
+				// if post draining has finished (job completed) remove the pod
+				drainedPost, err := isNodeDrained(c, node.Name, node.Name+"-drain-job-post", instance)
+				if err != nil && errors.IsNotFound(err) {
+					log.Error(err, fmt.Sprintf("Post NodeDrain job errors.IsNotFound: %s", node.Name+"-drain-job-post"))
+					ResyncPeriod = 30 * time.Second
+					return nil
+				} else if err != nil {
+					return err
+				}
+				log.Info(fmt.Sprintf("NodeDrainPost status: %v ", drainedPost))
+				switch drainedPost {
+				case "succeeded":
+					log.Info(fmt.Sprintf("NodeDrainPost job succeeded: %s", node.Name+"-drain-job-post"))
+					err = deleteBlockerPodFinalizer(c, node.Name, instance)
+					if err != nil {
+						return err
+					}
+					for i, nodeToDelete := range instance.Spec.NodesToDelete {
+						if nodeToDelete.Name == node.Name {
+							if len(instance.Spec.NodesToDelete) == 1 {
+								instance.Spec.NodesToDelete = []computenodev1alpha1.NodeToDelete{}
+							} else {
+								instance.Spec.NodesToDelete = removeNode(instance.Spec.NodesToDelete, i)
+							}
+							err = c.Update(context.TODO(), instance)
+							if err != nil {
+								return err
+							}
+							break
 						}
-						break
 					}
+				case "active":
+					// return reconsil
+					log.Info(fmt.Sprintf("NodeDrainPost job active: %s", node.Name+"-drain-job-post"))
+					ResyncPeriod = 30 * time.Second
+					return nil
+				default:
+					// return reconsil, this can happen if the created POST job misses no status.Succeeded or status.Active information
+					ResyncPeriod = 30 * time.Second
+					return nil
 				}
-			} else {
+
+			case "active":
+				// return reconsil
+				log.Info(fmt.Sprintf("NodeDrainPre job active: %s", node.Name+"-drain-job-pre"))
+				ResyncPeriod = 30 * time.Second
```

Comment:
we should not modify the resyncPeriod. Before the controller was reacting to status changes on the draining pod (lines 97-100). We should now react on the draining jobs instead, so that once the status of the job is changed the reconcile is triggered)